### PR TITLE
Mark format! with must_use hint

### PR DESCRIPTION
**Pull request by aceArt-GmbH**
Uses unstable feature https://github.com/rust-lang/rust/issues/94745

Part of #126475

First contribution to rust, please let me know if the blessing of tests is correct
Thanks @bjorn3 for the help

### CODE REVIEW

**Comment by workingjubilee** [comment]
You need to fix all the doctests as well.

**Comment by workingjubilee** [comment]
It looks like clippy's tests need help too.

**Comment by workingjubilee** [comment]
My condolences on your decision to implement this (desirable) change.

File: library/alloc/src/fmt.rs [hunk 1]
```diff
@@ -12,19 +12,19 @@
...
-//!                                   //       100,
-//!                                   //       200,
-//!                                   //     )"
+//! let _ = format!("Hello {people}!");       // => "Hello Rustaceans!"
+//! let _ = format!("{} {}", 1, 2);           // => "1 2"
+//! let _ = format!("{:04}", 42);             // => "0042" with leading zeros
+//! let _ = format!("{:#?}", (100, 200));     // => "(
+//!                                           //       100,
+//!                                           //       200,
+//!                                           //     )"
```
Comment (tgross35) [inline library/alloc/src/fmt.rs:27]: Imo adding the `let _ =` is just some unnecessary noise in a commonly referenced module. Could we just `println!` instead? Or disable the warning in these docs?

Comment (aceArt-GmbH) [reply library/alloc/src/fmt.rs:27]: this example is about `format!` so we should be using it and not `println!`.
But we can disable the warning in this doc

**Pull request merged**